lib/Doctrine/DBAL/DriverManager.php

```diff
@@ -115,11 +115,16 @@ private function __construct()
      * <b>driverClass</b>:
      * The driver class to use.
      *
-     * @param mixed[]            $params       The parameters.
-     * @param Configuration|null $config       The configuration to use.
-     * @param EventManager|null  $eventManager The event manager to use.
+     * @param array{wrapperClass?: class-string<T>} $params
```

Comment:
Can we turn this the other way around and use `@psalm-param` for this definition in order to not distract PHPStan that doesn't understand this?

Reply:
I will check, but PHPStan and Psalm read each others' annotation

Reply:
As I feared, it does not work, I get the error messages I described at https://github.com/doctrine/dbal/pull/3977#discussion_r418601811

Reply:
> I will check, but PHPStan and Psalm read each others' annotation

If they read each other's annotations but don't interpret equally then what's the point in that and having the `psalm-` and `phpstan-` prefixes? The difference in interpretation shouldn't be solved here.

Reply:
I think the point is not to confuse other tools like phpcs+slevomat, and that they started with not interpreting each other's rules at the beginning.